hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -181,16 +181,28 @@ def integers(
 )
 
 
+# Smallest positive non-zero numbers that is fully representable by an
+# IEEE-754 float, calculated with the width's associated minimum exponent.
+# Values from https://en.wikipedia.org/wiki/IEEE_754#Basic_and_interchange_formats
+width_smallest_normals = {
+    16: 2 ** -14,
+    32: 2 ** -126,
+    64: 2 ** -1022,
```

Comment:
I'd consider writing these with the number of exponent bits, which is a bit less magical to me: `64: 2 ** -(2 ** 11 - 2)  # eleven-bit exponent`.

We should also follow it immediately with `assert sys.float_info.min == width_smallest_normals[64]`, which is a nice sanity-check and will confirm to readers that we got the equation right.